utils/controller.js

```diff
@@ -0,0 +1,296 @@
+
+const Company = require('../model/company');
+const News = require('../model/news');
+const User = require('../model/user');
+const parameter = require('../utils/parameters');
+module.exports={ 
+    
+getAllUser(req,res){
+  User.find({},(err,data)=>{
+      return res.json({Customers : data});
+  })
+},
+    
+getAllCompanies(req,res){
+   Company.find({},(err,companies)=>{
+    if(err) res.json(err); 
+    res.json(companies);
+   })
+},
+
+getCompany(req,res){
+    Company.findById(req.params.id,(err,data)=>{
+        if(err) res.json(err);
+        res.json({compDetails:data});
+    })
+},
+
+getAllNews(req,res){
+    News.find({},(err,news)=>{
+        if(err) res.json(err);
+        res.json(news)
+    })
+},
+
+getNewsDetail(req,res){
+    News.findById(req.params.id,(err,data)=>{
+    if(err) res.json(err);
+     res.json(data);
+    })
+},
+
+getCustomerDetail(req,res){
+    User.findById(req.params.id,(err,data)=>{
+        res.json({Customer:data});
+    })
+},
+
+buystock(req,res){
+
+    let user ;
+    User.findById(req.body.id,(err,data)=>{
+        if(err) throw err;
+        user = data;
+        let total =0;
+    for(var i in user.stockHolding)
+    {
+        total += i.quantity;
+    }
+    for(var i in user.stockShorted)
+    {
+        total += i.quantity;
+    }
+    if(total+req.body.NoOfShares>parameter.maxShares)
+    {
+        return res.json({msg :"kitne kareedega?"});
+    }
+    })
+    
+    Company.findById(req.body.companyId,(err,companys)=>{
+        if(err) res.json(err);
+        if(user.accountBalance<companys.sharePrice*req.body.NoOfShares){
+           return res.json({msg :"itne paise nhi he"});
+        }
+        else{
+            if(companys.availableQuantity<req.body.NoOfShares)
+            {
+                return res.json({msg :"itne stock nhi he"});
+            }
+            var historytemp  = {
+                sharePrice : companys.sharePrice,
+                availableQuantity : companys.availableQuantity-req.body.NoOfShares
+            }
+            var activitytemp ={
+                company : companys._id,
+                action : "bought" ,
+    	        quantity : req.body.NoOfShares,
+    	        price : companys.sharePrice
+            }
+            companys.history.push(historytemp);
+            user.accountBalance -= companys.sharePrice*req.body.NoOfShares;
+            companys.availableQuantity -= req.body.NoOfShares;
+            user.activity.push(activitytemp);
+            if(user.stockHolding.id(companys._id))
```

Comment:
what does this check?

Also, replace companys with company, since you're finding just one company

Reply:
this checks if user has shares of that company